// yoomoney/lib/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _regenerator = _interopRequireDefault(require("@babel/runtime/regenerator"));

var _asyncToGenerator2 = _interopRequireDefault(require("@babel/runtime/helpers/asyncToGenerator"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _axios = _interopRequireDefault(require("axios"));

var _qs = _interopRequireDefault(require("qs"));

var Yoomoney = function Yoomoney(accessToken) {
  var _this = this;

  (0, _classCallCheck2["default"])(this, Yoomoney);
  (0, _defineProperty2["default"])(this, "sendAuthenticatedRequest", /*#__PURE__*/function () {
    var _ref2 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee(_ref) {
      var _ref$method, method, url, _ref$data, data, _ref$params, params, _yield$axios, response;

      return _regenerator["default"].wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              _ref$method = _ref.method, method = _ref$method === void 0 ? "POST" : _ref$method, url = _ref.url, _ref$data = _ref.data, data = _ref$data === void 0 ? {} : _ref$data, _ref$params = _ref.params, params = _ref$params === void 0 ? {} : _ref$params;
              _context.prev = 1;
              _context.next = 4;
              return (0, _axios["default"])({
                method: method,
                url: "https://yoomoney.ru".concat(url),
                headers: {
                  Authorization: "Bearer " + _this.accessToken
                },
                data: _qs["default"].stringify(data),
                params: params
              });

            case 4:
              _yield$axios = _context.sent;
              response = _yield$axios.data;
              return _context.abrupt("return", response);

            case 9:
              _context.prev = 9;
              _context.t0 = _context["catch"](1);

              if (!_context.t0.response) {
                _context.next = 15;
                break;
              }

              return _context.abrupt("return", {
                status: _context.t0.response.status,
                statusText: _context.t0.response.statusText,
                data: _context.t0.response.data
              });

            case 15:
              if (!_context.t0.request) {
                _context.next = 17;
                break;
              }

              return _context.abrupt("return", _context.t0.request);

            case 17:
              return _context.abrupt("return", _context.t0.message);

            case 18:
            case "end":
              return _context.stop();
          }
        }
      }, _callee, null, [[1, 9]]);
    }));

    return function (_x) {
      return _ref2.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "accountInfo", /*#__PURE__*/(0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee2() {
    return _regenerator["default"].wrap(function _callee2$(_context2) {
      while (1) {
        switch (_context2.prev = _context2.next) {
          case 0:
            _context2.next = 2;
            return _this.sendAuthenticatedRequest({
              url: "/api/account-info"
            });

          case 2:
            return _context2.abrupt("return", _context2.sent);

          case 3:
          case "end":
            return _context2.stop();
        }
      }
    }, _callee2);
  })));
  (0, _defineProperty2["default"])(this, "operationInfo", /*#__PURE__*/function () {
    var _ref4 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee3(data) {
      return _regenerator["default"].wrap(function _callee3$(_context3) {
        while (1) {
          switch (_context3.prev = _context3.next) {
            case 0:
              _context3.next = 2;
              return _this.sendAuthenticatedRequest({
                url: "/api/operation-history",
                data: data
              });

            case 2:
              return _context3.abrupt("return", _context3.sent);

            case 3:
            case "end":
              return _context3.stop();
          }
        }
      }, _callee3);
    }));

    return function (_x2) {
      return _ref4.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "operationDetails", /*#__PURE__*/function () {
    var _ref5 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee4(operation_id) {
      return _regenerator["default"].wrap(function _callee4$(_context4) {
        while (1) {
          switch (_context4.prev = _context4.next) {
            case 0:
              _context4.next = 2;
              return _this.sendAuthenticatedRequest({
                url: "/api/operation-details",
                data: {
                  operation_id: operation_id
                }
              });

            case 2:
              return _context4.abrupt("return", _context4.sent);

            case 3:
            case "end":
              return _context4.stop();
          }
        }
      }, _callee4);
    }));

    return function (_x3) {
      return _ref5.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "requestPayment", /*#__PURE__*/function () {
    var _ref6 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee5(data) {
      return _regenerator["default"].wrap(function _callee5$(_context5) {
        while (1) {
          switch (_context5.prev = _context5.next) {
            case 0:
              _context5.next = 2;
              return _this.sendAuthenticatedRequest({
                url: "/api/request-payment",
                data: data
              });

            case 2:
              return _context5.abrupt("return", _context5.sent);

            case 3:
            case "end":
              return _context5.stop();
          }
        }
      }, _callee5);
    }));

    return function (_x4) {
      return _ref6.apply(this, arguments);
    };
  }());
  (0, _defineProperty2["default"])(this, "processPayment", /*#__PURE__*/function () {
    var _ref7 = (0, _asyncToGenerator2["default"])( /*#__PURE__*/_regenerator["default"].mark(function _callee6(data) {
      return _regenerator["default"].wrap(function _callee6$(_context6) {
        while (1) {
          switch (_context6.prev = _context6.next) {
            case 0:
              _context6.next = 2;
              return _this.sendAuthenticatedRequest({
                url: "/api/process-payment",
                data: data
              });

            case 2:
              return _context6.abrupt("return", _context6.sent);

            case 3:
            case "end":
              return _context6.stop();
          }
        }
      }, _callee6);
    }));

    return function (_x5) {
      return _ref7.apply(this, arguments);
    };
  }());
  this.accessToken = accessToken;
};

exports["default"] = Yoomoney;
